rasa/core/processor.py

```diff
@@ -803,5 +802,5 @@ def _get_next_action_probabilities(
                 )
 
         return self.policy_ensemble.probabilities_using_best_policy(
-            tracker, self.domain
+            tracker, self.domain, self.interpreter
```

Comment:
I have a doubt here which I would be very happy to be wrong :) 
I think, with current setting, the self.interpreter will always be `RegexInterpreter`. it will be changed only if we're using the current e2e testing (I mean, this e2e: https://rasa.com/docs/rasa/user-guide/testing-your-assistant/).

Reply:
I don't think I can make a suggestion there but in `rasa/test.py` in `test_core` function removing the check for use_e2e should do the trick :)

Reply:
Haha, now I was bit confused myself 😆 The `interpreter` in processor is the same one which is already used for parsing the incoming NLU messages. So `self.interpreter` is definitely not the `RegexInterpreter`, but the one which is loaded by `Agent`

Reply:
Aaa 🤯
I am sorry to bother you with this but I need help finding flaws in my thinking :) 
So, e.g., when we do `rasa test core`, we get to `test_core` function [here](https://github.com/RasaHQ/rasa/blob/a706b30f86d343af26afb719e0447201eb9ff4e5/rasa/test.py#L146-L160). If `use_e2e` flag is false, `_interpreter` is `RegexInterpreter`  and that is passed to `Agent.load`.  And within `Agent.load` the new interpreter will not be loaded as we have passed the RegexInterpreter. 
(I really want to see where I'm wrong 🙈)

Reply:
You're right! I didn't think of testing - only of `rasa run`

Reply:
fixed and added a test